pyshelf/search/formatter.py

```diff
@@ -0,0 +1,104 @@
+from distutils.version import LooseVersion
+from pyshelf.search.sort_type import SortType
+from pyshelf.search.sort_flag import SortFlag
+from pyshelf.search.type import Type as SearchType
+
+
+class Formatter(object):
+    def __init__(self, criteria, search_results, key_list=None):
+        """
+            Formats search results from Elasticsearch based on the given criteria.
+
+            Args:
+                criteria(dict): This contains the search and sort criteria as outlined on
+                                pyshelf.search.manager.Manager.search.
+                search_results(List[elasticsearch_dsl.result.Result]): List of search results.
+                key_list(list): list of keys to include in filtered results if list is not passed
+                                all fields will be returned.
+        """
+        self.search_criteria = criteria.get("search")
+        # Sort criteria must be reversed to ensure first sort criteria takes precedence
+        self.sort_criteria = reversed(criteria.get("sort", []))
+        self.key_list = key_list
+        self.search_results = search_results
+        self._results = None
+        self.tilde_search = {}
+
+    @property
+    def results(self):
+        """
+            Filtered and sorted results.
+        """
+        self._results = self._sort_results(self._filter_results())
+        return self._results
+
+    def _filter_version_search(self, item_name, item_value):
+        """
+            This ensures when a version search is done that any results that are
+            less then the version that is passed in are dropped from the result set.
+
+            Args:
+                item_name
+        """
+        if not self.tilde_search:
+            for criteria in self.search_criteria:
+                if criteria["search_type"] == SearchType.VERSION:
+                    self.tilde_search[criteria["field"]] = criteria["value"]
+
+        return item_name in self.tilde_search.keys() and \
```

Comment:
This is a lot on a single line.  Can you split this up.  Also you shouldn't need to say in .keys().  in automatically works with dicts.

```
version_value = self.tilde_search.get(item_name)
is_bigger_or_whatever = False
if version_value:
    version_value = LooseVersion(version_value)
    item_value = LooseVersion(item_value)
    is_bigger_or_whatever = (item_value < version_value)

return is_bigger_or_whatever
```